src/metabase/query_processor/parameters.clj

```diff
@@ -10,18 +10,116 @@
             [metabase.util :as u])
   (:import (org.joda.time DateTimeConstants DateTime)))
 
+(defn- day-range
+  [^DateTime end ^DateTime start]
+  {:end   end
+   :start start})
 
-(def ^:private ^:const relative-dates
-  #{"today"
-    "yesterday"
-    "past7days"
-    "past30days"
-    "thisweek"
-    "thismonth"
-    "thisyear"
-    "lastweek"
-    "lastmonth"
-    "lastyear"})
+(defn- week-range
+  [^DateTime end ^DateTime start]
+    ;; weeks always start on SUNDAY and end on SATURDAY
+    ;; NOTE: in Joda the week starts on Monday and ends on Sunday, so to get the right Sunday we rollback 1 week
+   {:end   (.withDayOfWeek end DateTimeConstants/SATURDAY)
+    :start (.withDayOfWeek ^DateTime (t/minus start (t/weeks 1)) DateTimeConstants/SUNDAY)})
+
+(defn- month-range
+  [^DateTime end ^DateTime start]
+  {:end   (t/last-day-of-the-month end)
+   :start (t/first-day-of-the-month start)})
+
+(defn- year-range
+  [^DateTime end ^DateTime start]
+  {:end   (t/last-day-of-the-month  (.withMonthOfYear end DateTimeConstants/DECEMBER))
+   :start (t/first-day-of-the-month (.withMonthOfYear start DateTimeConstants/JANUARY))})
+
+(def ^:private operations-by-date-unit
+  {"day"   {:unit-range     day-range
+            :to-period t/days}
```

Comment:
looks like the indentation for the values of `:unit-range` got messed up a little here